src/bigunit.ts

```diff
@@ -399,27 +417,89 @@ export class BigUnit {
     // If the length is less than the precision, pad with zeros
     if (length < this.precision) {
       const padding = "0".repeat(this.precision - length);
-      return `0.${padding}${valueString}`;
+      // If the value is negative, add a negative sign to the padded zeros and remove the negative sign from the value
+      // This avoids results like "0.000-1234"
+      return valueString.includes("-")
+        ? `-0.${padding}${valueString.replace("-", "")}`
+        : `0.${padding}${valueString}`;
     }
 
     // If the length is greater than the precision, slice the string
     const decimalIndex = length - this.precision;
     const integerPart = valueString.slice(0, decimalIndex);
     const fractionalPart = valueString.slice(decimalIndex);
-    return `${integerPart}.${fractionalPart}`;
+    // If the integer part is empty, return "0" instead of an empty string
+    return `${integerPart === "" ? "0" : integerPart}.${fractionalPart}`;
   }
 
   public toBigInt(): bigint {
     return this.value;
   }
 
+  // /**
+  //  * @description Format the value as a string with the given precision (Note: this uses standard rounding)
+  //  * @param precision
+  //  * @returns string representation of the unit value
+  //  */
+  // public format(precision: number): string {
+  //   // TAYLOR OVERFLOW ERROR HERE!
+  //   return this.toNumber().toFixed(precision);
+  // }
+
   /**
-   * @description Format the value as a string with the given precision
-   * @param precision
+   * @description Format the value as a string with the given precision (Note: this uses standard rounding)
+   * @param precision - the number of digits to appear after the decimal point
    * @returns string representation of the unit value
    */
   public format(precision: number): string {
-    return this.toNumber().toFixed(precision);
+    let valueStr = this.toBigInt().toString();
```

Comment:
This seems really complex. Can we use the `toString()` method as a starting point?

Reply:
Sure, I'll have to do `let valueStr = this.toString().replace(".", "");` though